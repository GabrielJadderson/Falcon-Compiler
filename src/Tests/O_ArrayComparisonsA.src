# Code generation for the comparison operator and array assignment
# array vs. array and array vs. null
type myArr = array of int;

var a : myArr;
var b : myArr;
var temp : myArr;

allocate a of length 20;
allocate b of length 10;

a[19] = 19 * 10;
b[9] = 9 * 10;

write a[19];
write b[9];

temp = a;
a = b;
b = temp;

write a[9];
write b[19];

if( b == temp ) then write 1; else write 0;
b = null;
if( b == temp ) then write 1; else write 0;

