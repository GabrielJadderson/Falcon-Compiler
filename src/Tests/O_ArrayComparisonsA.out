190
90
90
190
1
0
